src/compiler/emitter.ts

```diff
@@ -4301,6 +4301,21 @@ namespace ts {
                                 shouldDecreaseIndentAfterEmit = true;
                             }
 
+                            // Emit comments for comma-delimited list nodes.
+                            // i.e
+                            //       var array = [...
+                            //          2, /* comment */
+                            //          ...
+                            //       ];
+                            if (mayEmitInterveningComments && format & ListFormat.CommaDelimited) {
+                                if (emitTrailingCommentsOfPosition) {
+                                    if (child.kind !== SyntaxKind.PropertyAssignment) {

```

Comment:
Why do PropertyAssignments need to be filtered out?

Also nit: these ifs can be collapsed into one.

Reply:
Duplicate comments were being emitted before filtering out PropertyAssignments. I believe PropertyAssignment comments are already being emitted by `emit(child)`, which is why it would make sense to filter here. But it does feel like a workaround, and I'm new to the TypeScript codebase so I'm unsure of a more correct way to go about this is.


And thanks for the suggestion on the if-statements, I will fix that shortly.

Reply:
@rbuckton will likely have a suggestion about whether there’s a better place for this to go where it wouldn’t have to filter like this, or maybe a better condition to filter by. I’m not very familiar with comment emit myself.